connectors/spark/src/test/java/io/unitycatalog/spark/auth/AwsCredentialRenewalTest.java

```diff
@@ -0,0 +1,362 @@
+
+                    // Post-check after the credential renewal.
+                    fs.getFileStatus(locPath);
+                    assertThat(fs.renewalCount()).isEqualTo(refreshIndex + 1);
+                  }
+
+                  return RowFactory.create(10);
+                })
+            .collect();
+
+    assertThat(rows.stream().map(r -> r.getInt(0)).collect(Collectors.toList()))
+        .isEqualTo(ImmutableList.of(10));
+  }
+
+  @Test
+  public void testDeltaReadWriteRenewal() throws Exception {
+    String location = String.format("s3://%s%s/delta", BUCKET_NAME, dataDir.getCanonicalPath());
+    StructType schema =
+        new StructType(
+            new StructField[] {
+              new StructField("id", DataTypes.IntegerType, false, Metadata.empty()),
+              new StructField("partition", DataTypes.IntegerType, false, Metadata.empty()),
+            });
+
+    // Create the external delta path-based table.
+    S3CredFileSystem.credentialCheckEnabled = false;
+    sql("CREATE TABLE delta.`%s` (id INT) USING delta PARTITIONED BY (partition INT)", location);
+    S3CredFileSystem.credentialCheckEnabled = true;
+
+    // Create the catalog table referring to external table.
+    sql("CREATE TABLE %s USING delta LOCATION '%s'", TABLE_NAME, location);
+
+    // Insert 4 rows into each partition where (id % 3) equals the partition key.
+    sql("INSERT INTO %s PARTITION (partition=1) VALUES (1), (4), (7), (10)", TABLE_NAME, 1);
+    sql("INSERT INTO %s PARTITION (partition=2) VALUES (2), (5), (8), (11)", TABLE_NAME, 2);
+    sql("INSERT INTO %s PARTITION (partition=3) VALUES (3), (6), (9), (12)", TABLE_NAME, 3);
+
+    // Read from the Delta table, mapping each partition to a separate task.
+    // With parallelism set to 1, tasks for each partition execute sequentially.
+    // The accumulated 30-second delay advances the clock sufficiently to trigger a
+    // renewal of filesystem credentials.
+
+    // The spark job should be success because we've enabled the credential renewal.
+    session
+        .read()
+        .format("delta")
+        .table(TABLE_NAME)
+        .mapPartitions(
+            (MapPartitionsFunction<Row, Row>)
+                input -> {
+                  // Advance the clock to trigger the credential renewal.
+                  testClock().advance(Duration.ofMillis(DEFAULT_INTERVAL_MILLIS));
+                  return input;
+                },
+            Encoders.row(schema))
+        .write()
+        .format("delta")
+        .mode("append")
+        .saveAsTable(TABLE_NAME);
+
+    List<Row> rows = sql("SELECT * FROM %s ORDER BY id ASC", TABLE_NAME);
+    assertThat(rows.size()).isEqualTo(24);
+    assertThat(rows.stream().map(r -> r.getInt(0)).collect(Collectors.toList()))
+        .isEqualTo(
+            IntStream.concat(IntStream.rangeClosed(1, 12), IntStream.rangeClosed(1, 12))
+                .boxed()
+                .sorted()
+                .collect(Collectors.toList()));
```

Comment:
general question: which test in the set of tests we have written till now ensures that the credentials are not accidentally used across queries even if they are for the same table?